chain/chain/src/chain.rs

```diff
@@ -435,7 +437,7 @@ impl Chain {
             blocks_delay_tracker: BlocksDelayTracker::new(clock.clone()),
             apply_chunks_sender: sc,
             apply_chunks_receiver: rc,
-            apply_chunks_spawner: Arc::new(RayonAsyncComputationSpawner),
+            apply_chunks_spawner: ApplyChunksSpawner::default().into_spawner(num_shards),
```

Comment:
I still think it's a bad idea to error on the limit at num shards. I haven't tested but I believe it is possible for the node to be processing multiple blocks at the same time - because of forks.

Reply:
Might even be a good idea to add a test to see it happening though it may not be easy to craft such a test.

Reply:
As discussed offline, I will change that into a soft limit and print warnings when it's exceeded.